connectors/services/job_execution.py

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+from functools import cached_property
+
+from connectors.es.client import License
+from connectors.es.index import DocumentNotFoundError
+from connectors.es.license import requires_platinum_license
+from connectors.logger import logger
+from connectors.protocol import (
+    ConnectorIndex,
+    DataSourceError,
+    JobType,
+    SyncJobIndex,
+)
+from connectors.services.base import BaseService
+from connectors.source import get_source_klass
+from connectors.sync_job_runner import SyncJobRunner
+from connectors.utils import ConcurrentTasks
+
+
+class JobExecutionService(BaseService):
+    name = "execute"
+    display_name = "job execution"
+    JOB_TYPES = [
+        JobType.ACCESS_CONTROL.value,
+        JobType.FULL.value,
+        JobType.INCREMENTAL.value,
+    ]
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.idling = self.service_config["idling"]
+        self.source_list = config["sources"]
+        self.sync_job_pool = ConcurrentTasks(max_concurrency=self.max_concurrency)
+
+    def stop(self):
+        super().stop()
+        self.sync_job_pool.cancel()
+
+    @cached_property
+    def max_concurrency(self):
+        raise NotImplementedError()
+
+    def should_execute(self, connector, sync_job):
+        return False
+
+    async def _sync(self, sync_job):
+        if sync_job.service_type not in self.source_list:
+            msg = f"Couldn't find data source class for {sync_job.service_type}"
+            raise DataSourceError(msg)
+        source_klass = get_source_klass(self.source_list[sync_job.service_type])
+        connector_id = sync_job.connector_id
+
+        try:
+            connector = await self.connector_index.fetch_by_id(connector_id)
+        except DocumentNotFoundError:
+            sync_job.log_error("Couldn't find connector")
+            return
+
+        if requires_platinum_license(sync_job, connector, source_klass):
+            (
+                is_platinum_license_enabled,
+                license_enabled,
+            ) = await self.connector_index.has_active_license_enabled(License.PLATINUM)
+
+            if not is_platinum_license_enabled:
+                sync_job.log_error(
+                    f"Minimum required Elasticsearch license: '{License.PLATINUM.value}'. Actual license: '{license_enabled.value}'."
+                )
+                return
```

Comment:
Probably not for this PR, but I'd move this logic around a bit so that we would not show this error if the job is supposed to execute.

Cause now if you're using connector that supports permissions, but have no license + no permissions enabled you still see this error in console.

Should I create an issue for it?

Reply:
In the happy path (not considering users manually index a sync job document), this can happen in a very rare case, when the license expires after the acl sync job is scheduled/manually triggered (We also check licence when scheduling an acl sync job). 

When it happens, we have no way to mark this job as error and the same job will be picked up in every iteration. I think we should move this licence check into `SyncJobRunner`, which will mark the job as error if it doesn't have the minimum licence required.

I have created the issue: https://github.com/elastic/connectors/issues/1924